offchain-modules/src/packages/ckb/tx-helper/generator.ts

```diff
@@ -1,29 +1,336 @@
+      outputsData,
+    };
+    console.dir({ rawTx }, { depth: null });
+    return rawTx;
   }
 
-  async createBridgeCell(fromLockscript: Script, bridgeLockscriptArgs: string | Asset): Promise<Transaction> {
-    throw new Error('not implemented');
+  async mint(userLockscript: Script, records: MintAssetRecord[]): Promise<CKBComponents.RawTransactionToSign> {
+    logger.debug('start to mint records: ', records);
+    const bridgeCells = new Array(0);
+    const outputs = new Array(0);
+    const outputsData = new Array(0);
+    const sudtCellCapacity = 300n * 10n ** 8n;
+    for (const record of records) {
+      const recipientLockscript = record.recipient.toLockScript();
+      const bridgeCellLockscript = {
+        codeHash: ForceBridgeCore.config.ckb.deps.bridgeLock.script.codeHash,
+        hashType: ForceBridgeCore.config.ckb.deps.bridgeLock.script.hashType,
+        args: record.asset.toBridgeLockscriptArgs(),
+      };
+      const searchKey = {
+        script: new Script(
+          bridgeCellLockscript.codeHash,
+          bridgeCellLockscript.args,
+          bridgeCellLockscript.hashType,
+        ).serializeJson() as LumosScript,
+        script_type: ScriptType.lock,
+      };
+      const cells = await this.collector.indexer.getCells(searchKey);
+      const bridgeCell = cells[0];
+      const sudtArgs = this.ckb.utils.scriptToHash(<CKBComponents.Script>bridgeCellLockscript);
+      const outputSudtCell = {
+        lock: recipientLockscript,
+        type: {
+          codeHash: ForceBridgeCore.config.ckb.deps.sudtType.script.codeHash,
+          hashType: ForceBridgeCore.config.ckb.deps.sudtType.script.hashType,
+          args: sudtArgs,
+        },
+        capacity: `0x${sudtCellCapacity.toString(16)}`,
+      };
+      const outputBridgeCell = {
+        lock: bridgeCellLockscript,
+        capacity: bridgeCell.capacity,
+      };
+      outputs.push(outputSudtCell);
+      outputsData.push(record.amount.toUInt128LE());
+      outputs.push(outputBridgeCell);
+      outputsData.push('0x');
+      bridgeCells.push(bridgeCell);
+    }
+
+    const fee = 100000n;
+    // const inputCells = bridgeCells;
+    const needSupplyCap = sudtCellCapacity * BigInt(records.length) + fee;
+    const supplyCapCells = await this.collector.getCellsByLockscriptAndCapacity(
+      userLockscript,
+      Amount.fromUInt128LE(bigintToSudtAmount(needSupplyCap)),
+    );
+    const inputCells = supplyCapCells.concat(bridgeCells);
+    const inputs = inputCells.map((cell) => {
+      return { previousOutput: cell.outPoint, since: '0x0' };
+    });
+    this.handleChangeCell(inputCells, outputs, outputsData, userLockscript, fee);
+
+    const { secp256k1Dep } = await this.ckb.loadDeps();
+    const cellDeps = [
+      {
+        outPoint: secp256k1Dep.outPoint,
+        depType: secp256k1Dep.depType,
+      },
+      // sudt dep
+      {
+        outPoint: ForceBridgeCore.config.ckb.deps.sudtType.cellDep.outPoint,
+        depType: 'code',
```

Comment:
There is `depType` field in the config, should not hardcode it.

Reply:
fixed